getnacelle/nacelle-js
# refactor(vue): use Warp 2 shape; improve testability

### WHY are these changes introduced?

Addresses [LS-1214](https://nacelle.atlassian.net/browse/LS-1214)

> Update Refinement Provider with Warp 2 API schema shape



### WHAT is this pull request doing?



- Updates `<RefinementProvider />` and mocks to use Warp 2 data shapes.
- Refactors `<RefinementProvider />` logic into `src/utils/refinement` utility functions for improved testability.
- Refactors Web Workers used in `<RefinementProvider />` and `<SearchProvider />` to take advantage of [Vite's native support for Web Workers](https://vitejs.dev/guide/features.html#web-workers). 
  - Previously, we were converting functions into [Blobs](https://developer.mozilla.org/en-US/docs/Web/API/Blob), which made those functions very difficult to test.
- Adds test assertions to validate filtering & sorting logic.
- Updates all Vite-built `nacelle-js` projects to use latest `vite`

## 

### How to Test

1. Check out the `LS-1214-warp2-refinement-provider` branch
2. `cd packages/vue` 
3. `npm run test:ci` (all tests should pass)
4. `npm run build` (it should build `es`, `umd`, and `iife` bundles with `assets/` containing four Web Workers):
<img width="240" alt="@nacelle:vue build outputs" src="https://user-images.githubusercontent.com/5732000/146221637-e52ed979-bf8e-4b57-929d-3a9b8928bee5.png">

Note that when building, you'll likely see a warning:

> If you do not supply "output.name", you may not be able to access the exports of an IIFE bundle.

While I'm not sure why that warning appears when bundling workers in `dist/assets`, we are specifying the `name` in `build.lib` correctly, as described [in the Vite docs](https://vitejs.dev/guide/build.html#library-mode) and as demonstrated [in `petite-vue`](https://github.com/vuejs/petite-vue/blob/63be8b5ade108f24a1f8d00bc7230507d19d0a82/vite.config.ts#L8-L12).

### Next Steps

We will need to wait for Warp 2 `starters/nuxt` updates ([LS-1294](https://nacelle.atlassian.net/browse/LS-1294)) before we can test these changes in `starters/nuxt`.
